jekyll/_cci2/security-recommendations.md

```diff
@@ -21,16 +21,108 @@ If you are getting started with CircleCI, there are some security best practices
+
+If writing a shell script that uses the CircleCI CLI, remember that in Bash you can avoid exposing secrets stored in environment variables or text by using the `<<<` construct, which does not spawn a new process while piping a value: `circleci context store-secret <vcs-type> <org-name> <context-name> <secret name> <<< "$MY_SECRET"`. This is more reliable than using `echo` or `printf`, which may or may not be shell builtins and therefore could spawn a process.
+
+#### Using curl safely with the CircleCI API
+{: #using-curl-safely-with-the-circleci-api }
+
+##### Protecting the API token
+{: #protecting-the-api-token }
+
+When calling the CircleCI API with `curl`, you need to provide an API token. There are several ways you can mitigate risk while doing so:
+
+* Use a `.netrc` file: The [netrc file format](https://everything.curl.dev/usingcurl/netrc), which is supported by several different tools, allows you to provide HTTP basic auth credentials in a file, rather than at the command-line.
+  - Create a file at a location of your choosing; the default used by some tools is `~/.netrc`. Be sure to `chmod 0600` this file before adding the secret, to prevent other users from viewing its contents.
+  - Add a line in the following format: `machine circleci.com login <your token> password`
+  - When invoking `curl`, tell it to look in your `.netrc` file for credentials: `curl --netrc-file ~/.netrc`
+* Write the `Circle-Token` header into a file. This requires curl 7.55 or later, but is a more reliable solution than `.netrc`, because some tools that use `.netrc` files do not understand an empty password field.
+  - Create a file at a location of your choosing; be sure to chmod 0600 the file to prevent other users from viewing its contents.
+  - Add a line in the following format: `Circle-Token: <your token>`
+  - When invoking `curl`, tell it to read the header from a file: `curl --header @your_filename`
+* Pull the token directly from a tool designed to store secrets, such as 1Password. In this case, you can use [process substitution](https://en.wikipedia.org/wiki/Process_substitution) to retrieve the header without exposing it:
+  - `curl --header @<(command_to_retrieve_password)`
+  - If you are sure that `printf` is a builtin in your shell, it should also be safe to write `curl --header @<(printf '%s\n' "$MYVAR")`, allowing you to use environment variables without exposing them through ps.
+
+##### Protecting your secrets
+{: #protecting-your-secrets }
+
+Some API endpoints, such as [addEnvironmentVariableToContext]({{site.baseurl}}/api/v2/#operation/addEnvironmentVariableToContext), may require secrets to be sent in the body of `PUT` or `POST` requests. There are several options to help conceal these secrets.
+
+* Use a file to compose and store the request body. Be sure to `chmod 0600` this file before adding the secret value, to prevent other users from viewing its contents.
```

Comment:
```suggestion
* Use a file to compose and store the request body. Be sure to `chmod 0600` this file before adding the secret value to prevent other users from viewing its contents.
```